### ISSUE TITLE
ValueError: cannot mmap an empty file

### ISSUE THREAD

**Issue by xiang-xiang-zhu** [ShannonAI/OpenViDial#37]
When I want to view the shape of train.features.mmap, numpy reports an error. How can I solve this problem

By the way, can I directly use the mmap file (such as train/valid/test. features.mmap) as the video feature, for example, save it as an .npy file for multimodal training

thank you

**Comment by xiang-xiang-zhu** [comment]
Because I didn't download the complete image compression package, I want to know how many images there are in the training set, verification set and test set respectively

**Comment by YuxianMeng** [comment]
@xiang-xiang-zhu Hi, I guess you are not familiar with [mmap format](https://numpy.org/doc/stable/reference/generated/numpy.memmap.html).
We choose to use `mmap` instead of `npy` is because `.npy` will load `np.array` to memory, but our feature file is too
big. 
If you want to know the number of images without downloading them, you can try to download text first, and each sentence in text should be paired with an image.

**Comment by xiang-xiang-zhu** [comment]
Now I want to train with my own model. Can I directly use MMAP to read the data in the image part? Is the shape (image_num,feature_ dim)?

**Comment by YuxianMeng** [comment]
@xiang-xiang-zhu Yes, please refer to our [corresponding code](https://github.com/ShannonAI/OpenViDial/blob/main/video_dialogue_model/data/feature_dataset.py#L29).

**Comment by xiang-xiang-zhu** [comment]
> @xiang-xiang-zhu Yes, please refer to our [corresponding code](https://github.com/ShannonAI/OpenViDial/blob/main/video_dialogue_model/data/feature_dataset.py#L29).

thank you very much
I would like to ask whether [0, 1, 2] in the jsonl file represents a series of conversations from the 0 to the 2 sentences. 1 is the response of 0 and 2 is the response of 1. The conversations in [3,4,5] have nothing to do with [0,1,2] during training

**Comment by YuxianMeng** [comment]
@xiang-xiang-zhu Yes, both of your comments are right.
